src/app/@ansyn/imagery/cache-service/cache.service.ts

```diff
@@ -24,4 +24,9 @@ export class CacheService {
 		}
 		this.cachedLayesrMap.set(id, [...layers]);
 	}
+
+	removeLayerFromCache(layer: any) {
```

Comment:
service shouldn't know how to generate id.. 
```typescript
removeLayerFromCache(id: string) {
}
```
or create public functions:

```typescript 
generateIdByLayer() ...
getLayerById() ...

```
something like this...